Sources/PubNub/Helpers/Crypto/Cryptors/LegacyCryptor.swift

```diff
@@ -0,0 +1,184 @@
+//
+//  LegacyCryptoAlgorithm.swift
+//
+//  PubNub Real-time Cloud-Hosted Push API and Push Notification Client Frameworks
+//  Copyright © 2023 PubNub Inc.
+//  https://www.pubnub.com/
+//  https://www.pubnub.com/terms
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+import Foundation
+import CommonCrypto
+
+/// Provides backward-compatible way of encryption/decryption that matches
+/// deprecated [Crypto](https://github.com/pubnub/swift/blob/master/Sources/PubNub/Helpers/Crypto/Crypto.swift#L32)
+///
+/// - Important: Using this `Cryptor` for encoding is strongly discouraged. Use ``AESCBCCryptor`` instead.
+public struct LegacyCryptor: Cryptor {
+  private let key: Data
+  private let withRandomIV: Bool
+  
+  static let ID: CryptorId = [0x00, 0x00, 0x00, 0x00]
+  
+  public init(key: String, withRandomIV: Bool = true) {
+    let hash = CryptorUtils.SHA256.hash(from: key.data(using: .utf8) ?? Data())
+    let hexStrData = CryptorUtils.hexFrom(hash).lowercased(with: .current).data(using: .utf8) ?? Data()
+    self.key = hexStrData
+    self.withRandomIV = withRandomIV
+  }
+  
+  public var id: CryptorId {
+    Self.ID
+  }
+  
+  public func encrypt(data: Data) -> Result<EncryptedData, Error> {
+    do {
+      let vectorGen = withRandomIV ? CryptorVector.random(bytesCount: kCCBlockSizeAES128) : CryptorVector.fixed
+      let ivData = try vectorGen.data()
+      
+      let encrypted = try data.crypt(
+        operation: CCOperation(kCCEncrypt),
+        algorithm: CCAlgorithm(kCCAlgorithmAES128),
+        options: CCOptions(kCCOptionPKCS7Padding),
+        blockSize: kCCBlockSizeAES128,
+        key: key,
+        initializationVector: ivData,
+        messageData: data
+      )
+      
+      // Join IV and encrypted content when using a random IV
+      return .success(EncryptedData(
+        metadata: ivData,
```

Comment:
Wouldn't be any issues here if we use random vector, then it will be included both in `metadata` and `data`?
Legacy cryptor doesn't have a header, so there is no metadata for it and all information stored right in the `data` field.